### PR TITLE
Test/gh pages

### PR DESCRIPTION
**Pull request by BenBradnick**
Testing creating a version file for the gh-pages branch based on the available builds.

### CODE REVIEW

File: docs/github_pages_version_file.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    tags = get_sorted_tags_list()
+
+    # Get the directories (i.e. builds) from the GitHub Pages branch
+    all_build_versions = get_branch_contents(remote, github_pages_branch)
+
+    # Parse all builds, storing releases in an unordered dict
+    unsorted_releases: Dict[str, GitHubPagesBuild] = {}
+    for version in all_build_versions:
+        # Create our instance
+        url = get_docs_url(github_org, project, version)
```
Comment (thomascobb) [inline docs/github_pages_version_file.py:151]: Don't need the URL, just the name

File: docs/github_pages_version_file.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        if status:
+            print(f"File {filename} being commited to {remote}/{branch}.")
+            # Add the file
+            run_command(f"git add {filename}")
+
+            # Commit the file
+            run_command("git commit -qm Update-version-file")
+
+            # Push to remote
+            run_command("git push -q")
```
Comment (thomascobb) [inline docs/github_pages_version_file.py:114]: We have permissions to do this? That makes things easier!

Comment (thomascobb) [reply docs/github_pages_version_file.py:114]: Could we do the deploy of the build/html directory too then? Save our use of the external action:
https://github.com/dls-controls/scanspec/blob/test/gh-pages/.github/workflows/docs.yml#L41-L60

**Review by thomascobb — COMMENTED** [review]
I think we can cut this down a bit, then we could make it a script in https://github.com/dls-controls/sphinx_rtd_theme_github_versions

File: docs/github_pages_version_file.py (hunk 1 above)
Comment (BenBradnick) [reply docs/github_pages_version_file.py]: Now just creates a text file of a list of versions: https://github.com/dls-controls/scanspec/blob/test/gh-pages-file/version.txt

File: docs/github_pages_version_file.py (hunk 2 above)
Comment (BenBradnick) [reply docs/github_pages_version_file.py:107]: Looks like it! I just ran this build which commits the file: https://github.com/dls-controls/scanspec/runs/4379379099?check_suite_focus=true

The file was committed to the test branch here: https://github.com/dls-controls/scanspec/blob/test/gh-pages-file/version.txt

**Comment by BenBradnick** [comment]
> I think we can cut this down a bit, then we could make it a script in https://github.com/dls-controls/sphinx_rtd_theme_github_versions

How should this be cut down and how does the script end up being called from the sphinx repo?

File: docs/github_pages_version_file.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    stdout = run_command(
+        f"git ls-tree -d --name-only {remote}/{branch}", return_stdout=True
+    )
+    assert isinstance(
+        stdout, str
+    ), f"Error getting directory contents of {remote}/{branch}"
+    return stdout.strip().split("\n")
+
+
+def get_docs_url(github_org: str, project: str, version: str):
```
Comment (thomascobb) [inline docs/github_pages_version_file.py:47]: This function is unused?

File: docs/github_pages_version_file.py [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        else:
+            branches.append(version)
+
+    # Use output from git tag to sort the releases
+    for tag in tags:
+        if tag in unsorted_releases:
+            releases.append(tag)
+
+    # Get version file string
+    version_file_string = get_version_file_string(main_branches, branches, releases)
```
Comment (thomascobb) [inline docs/github_pages_version_file.py:155]:
```suggestion
    version_file_string = "\n".join(main_branches + branches + releases)
```

File: docs/github_pages_version_file.py [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+    ), f"Error getting directory contents of {remote}/{branch}"
+    return stdout.strip().split("\n")
+
+
+def get_docs_url(github_org: str, project: str, version: str):
+    """Get the URL for a specific build of the project."""
+    return f"https://{github_org}.github.io/{project}/{version}"
+
+
+def get_version_file_string(
```
Comment (thomascobb) [inline docs/github_pages_version_file.py:52]: ...which would make this function unused

File: docs/github_pages_version_file.py (hunk 4 above)
Comment (BenBradnick) [reply docs/github_pages_version_file.py]: I always forget about join...

**Comment by thomascobb** [comment]
Code moved to https://github.com/dls-controls/sphinx_rtd_theme_github_versions

**Pull request closed without merging**